__tests__/components/TransactionList/TransactionRow.spec.ts

```diff
@@ -0,0 +1,888 @@
+            expect(amount).toBe(100);
+            expect(amountColor).toBe('green');
+            expect(hasNetworkMosaic).toBe(true);
+            expect(getAmountMosaicId.toHex()).toBe('091F837E059AE13C');
+            expect(isAmountShowTicker).toBe(false);
+        });
+
+        test('display message icon', () => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile },
+                {
+                    transaction: mockTransferTransaction(
+                        [new Mosaic(new MosaicId('461A9811A1DC1FBB'), UInt64.fromUint(0))],
+                        PlainMessage.create('test-message'),
+                    ),
+                },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act:
+            const hasMessage = vm.hasMessage();
+
+            // Assert:
+            expect(hasMessage).toBe(true);
+            expect(vm.messagePayload).toBe('test-message');
+            expect(wrapper.find('.extend-icon-holder.message').exists()).toBe(true);
+        });
+
+        test('hide message icon', () => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper({ currentProfile: testnetProfile }, { transaction: mockTransferTransaction() });
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act:
+            const hasMessage = vm.hasMessage();
+
+            // Assert:
+            expect(hasMessage).toBe(false);
+            expect(wrapper.find('.extend-icon-holder.message').exists()).toBe(false);
+        });
+
+        test('returns encrypted message title when message type is EncryptedMessage', () => {
+            // Arrange:
+            const recipient = getTestAccount('remoteTestnetRecipient');
+
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile },
+                {
+                    transaction: mockTransferTransaction(
+                        [],
+                        EncryptedMessage.create('test', recipient.publicAccount, currentSigner.privateKey),
+                    ),
+                },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act + Assert:
+            expect(vm.messagePayload).toBe('Encrypted message');
+        });
+
+        test('returns message payload in truncate when message length over 40', () => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile },
+                { transaction: mockTransferTransaction([], PlainMessage.create('0'.repeat(45))) },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act + Assert:
+            expect(vm.messagePayload).toBe('0000000000000000000000000000000000000000...');
+        });
+
+        test('display mosaic icon when mosaic excluded network mosaic', () => {
```

Comment:
what do you mean by "excluded" ?

Reply:
`excluded` may not be the right word here, basically, the mosaic icon will display when the mosaics list has other than network mosaic.

Reply:
what icon should be shown when both network and other mosaics are present?